NickvisionTubeConverter.GNOME/Blueprints/add_download_dialog.blp

```diff
@@ -1,150 +1,187 @@
-                valign: center;
-                icon-name: "folder-open-symbolic";
-                tooltip-text: _("SelectSaveFolder");
-        
-                styles ["flat"]
-              }
-            }
 
-            Adw.ActionRow _overwriteRow {
-              title: _("OverwriteExistingFiles");
-              activatable-widget: _overwriteSwitch;
+                    Adw.EntryRow _saveFolderRow {
+                      title: _("SaveFolder.Field");
+                      width-request: 420;
+                      editable: false;
+              
+                      [suffix]
+                      Gtk.MenuButton _saveWarning {
+                        valign: center;
+                        icon-name: "dialog-warning-symbolic";
+                        visible: false;
+                        popover: Gtk.Popover {
+                          Adw.Clamp {
+                            maximum-size: 300;
+                    
+                            Gtk.Label {
+                              label: _("SaveWarning.GTK");
+                              wrap: true;
+                              justify: center;
+                            }
+                          }
+                        };
+                
+                        styles ["flat", "warning"]
+                      }
+              
+                      [suffix]
+                      Gtk.Button _selectSaveFolderButton {
+                        valign: center;
+                        icon-name: "folder-open-symbolic";
+                        tooltip-text: _("SelectSaveFolder");
+                
+                        styles ["flat"]
+                      }
+                    }
 
-              [suffix]
-              Gtk.Switch _overwriteSwitch {
-                valign: center;
-                active: true;
-              }
-            }
-          }
+                    Adw.ActionRow _overwriteRow {
+                      title: _("OverwriteExistingFiles");
+                      activatable-widget: _overwriteSwitch;
 
-          Gtk.ToggleButton _numberVideosButton {
-            halign: start;
+                      [suffix]
+                      Gtk.Switch _overwriteSwitch {
+                        valign: center;
+                        active: true;
+                      }
+                    }
+                  }
+
+                  Gtk.ToggleButton _numberVideosButton {
+                    halign: start;
+
+                    Adw.ButtonContent {
+                        icon-name: "view-list-ordered-symbolic";
+                        label: _("NumberVideos");
+                    }
+                  }
 
-            Adw.ButtonContent {
-                icon-name: "view-list-ordered-symbolic";
-                label: _("NumberVideos");
+                  Adw.PreferencesGroup _videosGroup {
+                    margin-start: 2;
+                    margin-end: 2;
+                  }
```

Comment:
This should be wrapped in Gtk.ScrolledWindow, otherwise for long playlists the window will be incredibly big.

Reply:
The Adw.ViewStack is already scrollable oddly enough...

Reply:
> ~The Adw.ViewStack contents are already scrollable oddly enough...~
> 
> I've just noticed that the child widget for the download page is a Gtk.ScrolledWindow

Right, I missed that. But we don't want the whole page to be scrollable. At least the heading and the download button should always be visible to a user.

Reply:
> I see the button is outside of Gtk.ScrolledWindow. Still, I think it would be best to make only videos list scrollable.

Agreed. The videos list should be the only thing scrollable.

Reply:
Done, feel free to tweak